libs/dyn/convert/normalize.go

```diff
@@ -371,3 +373,8 @@ func (n normalizeOptions) normalizeFloat(typ reflect.Type, src dyn.Value, path d
 
 	return dyn.NewValue(out, src.Location()), diags
 }
+
+func (n normalizeOptions) normalizeInterface(typ reflect.Type, src dyn.Value, path dyn.Path) (dyn.Value, diag.Diagnostics) {
+	out := src.AsAny()
```

Comment:
We don't need to run `AsAny` here; it will convert the input value to the corresponding Go value.

This works now because the test uses primitive values, but if you change it to use a map or a slice, then I suspect this will start to fail (`dyn.NewValue` with an incompatible type). It should be safe to return `src` directly because we'll never have to convert it.